### ISSUE TITLE
Use `SFTTrainer` for completion-only model without `DataCollatorForCompletionOnlyLM`

### ISSUE THREAD

**Issue by dfdx** [huggingface/trl#1507]
Looking at the [Dataset format support](https://huggingface.co/docs/trl/en/sft_trainer#dataset-format-support) section of the tutorial, I read it as if using a dataset in instruction-following format is enough to tell `SFTTrainer` to calculate loss only on completions and not prompts. In other words, you either need `DataCollatorForCompletionOnlyLM` OR dataset in instruction-following format. 

However, from the code it looks like if we don't use `DataCollatorForCompletionOnlyLM`, then `DataCollatorForLanguageModeling` is used and the loss is calculated on both - prompt and completion. 

Do I misread the tutorial or the code? More generally, what is the simplest way to train on completions only?

**Comment by pretidav** [comment]
same question here

**Comment by yechenzhi** [comment]
same question here.

**Comment by younesbelkada** [comment]
Hi everyone,
Thanks for raising the question, reading this specific section: https://huggingface.co/docs/trl/en/sft_trainer#dataset-format-support for me it doesn't really tell that we do support out of the box training on completion only. If my understanding is correct, in your case you will need to explicitly pass `DataCollatorForCompletionOnlyLM` if you want to train on completions only using that dataset. SFTTrainer will first format your dataset correctly, then use that collator during training to train on completions only in an orthogonal way.

**Comment by YooSungHyun** [comment]
This is very strange. **SFT** refers to supervised fine-tuning, but when looking at the `SFTTrainer` code alone, it seems as if it was designed with **pretraining** in mind.  

If that's the case, users who use the example code without analyzing it (assuming that **SFT refers to instruct tuning**) may see unexpected training behavior and never realize what went wrong. Isn't this a serious issue?
